docs/extensibility.md

```diff
@@ -0,0 +1,94 @@
+# Extensibility
+
+Extensibility is key for WordPress and like the rest of WordPress components, Gutenberg is hightly extensible.
+
+
+## Creating Blocks
+
+Gutenberg is about blocks and the main extensibility API of Gutenberg is the Block API. It allows you to create static blocks, dynamic blocks rendering on the server and also blocks saving data to Post Meta for more structured content.
+
+If you want to learn more about block creations, The [Blocks Tutorial](./blocks) is the best place to sart.
+
+
+## Removing Blocks
+
+### Using a black list
+
+Adding blocks is easy enough, removing them is as easy. Plugin or theme authors have the possibility to "unregister" blocks.
+
+```js
+// myplugin.js
+
+wp.blocks.unregisterBlockType( 'core/verse' );
+```
+
+and load this script in the Editor
+
+```php
+<?php
+// myplugin.php
+
+function myplugin_blacklist_blocks() {
+	wp_enqueue_script(
+		'myplugin-blacklist-blocks',
+		plugins_url( 'myplugin.js', __FILE__ ),
+		array( 'wp-blocks' )
+	);
+}
+add_action( 'enqueue_block_editor_assets', 'myplugin_blacklist_blocks' );
+```
+
+
+### Using a white list
+
+If you want to disable all blocks except a whitelisted list, you can adapt the script above like so:
+
+```js
+// myplugin.js
+var allowedBlocks = [
+  'core/paragraph',
+  'core/image',
+  'core/html',
+  'core/freeform'
+];
+
+wp.blocks.getBlockTypes().forEach( function( blockType ) {
+  if ( allowedBlocks.indexOf( blockType.name ) === -1 ) {
+    wp.blocks.unregisterBlockType( blockType.name );
+  }
+} );
+```
+
+
+## Modifying Blocks
+
+To modify the behaviour of existing blocks, Gutenberg exposes a list of filters:
```

Comment:
This seems like it will be difficult to maintain (read: will become out of date), and would be nice if we could move to automate this, similar to documentation for core PHP.

Reply:
Yep good idea. Do you think we should land this as "Experimental" and iterate. Automating this might not be so easy, I don't know.